app/helpers/data_table_helper.rb

```diff
@@ -110,46 +115,93 @@ def render_data_table(data_table, options = {})
-              :column_picker_visible => column_picker_visible,
-              :download_visible => download_visible,
-              :xls_download_visible => xls_download_visible,
-              :has_s3_download => has_s3_download,
-              :reset_layout_visible => reset_layout_visible,
-              :converted_load_options => url_params.to_json,
-              :bulk_actions_visible => bulk_actions_visible,
-              :disable_state_storing => disable_state_storing,
-              :filter_form_id => filter_form_id,
-              :requireTotalRowCountIndicator => requireTotalRowCountIndicator,
-              :options => options,
-              :data_options_json => data_options_json,
-              :state_storing_json => state_storing_json
-            })
+
+    render(partial: 'data_tables/data_table',
+           locals: {
+             data_table: data_table,
+             container_id: container_id,
+             functions: functions,
+             height: height,
+             width: width,
+             options_json: options_json,
+             columns_json: columns_json,
+             compact_view_json: compact_view_json,
+             summaries_json: summaries_json,
+             custom_summary_functions: custom_summary_functions,
+             group_panel_visible: group_panel_visible,
+             column_picker_visible: column_picker_visible,
+             download_visible: download_visible,
+             xls_download_visible: xls_download_visible,
+             has_s3_download: has_s3_download,
+             reset_layout_visible: reset_layout_visible,
+             converted_load_options: url_params.to_json,
+             bulk_actions_visible: bulk_actions_visible,
+             disable_state_storing: disable_state_storing,
+             filter_form_id: filter_form_id,
+             requireTotalRowCountIndicator: requireTotalRowCountIndicator,
+             options: options,
+             data_options_json: data_options_json,
+             state_storing_json: state_storing_json
+           })
   end
 
   private
 
+  # To use this functionality, the following values must be set
+  # - call method in data_table -> store_source :filter_path (preferably in the base table)
+  # - set option :custom_filter => on data_table options Example: { :filter_class => Views::Composite.to_s, :use_query => { :query_class => CompositeQuery.to_s, :query_filter_class => CompositeQueryFilter.to_s } }
+  # - set use_query when you want to apply the existing permissions/conditions in the query_class
+  # - set filter_class as fallback if you don't care about the query_class advantages
+  # - set column option { :custom_filter => true }
+  # @param column Column
+  # @param data_table DataTable
+  def custom_filter(column, data_table)
+    query_class = nil
+    query_filter_class = nil
+    if data_table.options[:custom_filter][:use_query]
+      query_class = data_table.options[:custom_filter][:use_query][:query_class]
```

Comment:
I'm not ure if I'm reading this right butI would have expected this to be specified as a parameter to this method instead of fetching it from the data_table.options, this would allow customisation of these values at the column level and not just the data_table level.

i.e.
```ruby
#in the data table's for example
#instead of this in the columns
c.text              :FirmShortName, { :custom_filter => true }
c.text              :CurrencyName, { :custom_filter => true }
# and this on the data table
option :custom_filter => { :filter_class => Views::Composite.to_s, :use_query => { :query_class => CompositeQuery.to_s, :query_filter_class => CompositeQueryFilter.to_s } }

# you could allow this at column level only and not bother with the option?
c.text              :FirmShortName, { :custom_filter => { :filter_class => Views::Composite.to_s, :use_query => { :query_class => CompositeQuery.to_s, :query_filter_class => CompositeQueryFilter.to_s } } }
c.text              :CurrencyName, { :custom_filter => { :filter_class => SomeCurrencyModel.to_s, :use_query => { :query_class => SomeCurrencyModelQuery.to_s, :query_filter_class => SomeCurrencyModelQueryFilter.to_s } } }


```